crypto_analytics/collection/data_source/kraken.py

```diff
@@ -25,39 +29,57 @@ def fetch(self) -> pd.DataFrame:
         if not interval_int:
             raise ValueError('Interval must be daily, hourly or minute')
 
-        kraken = krakenex.API()
+        endpoint = 'https://api.kraken.com/0/public/OHLC'
 
-        parameters = {
+        last_time = time.time() if not self.last_time else self.last_time
+        interval_duration = self.interval.to_unix_time()
+        since = (last_time//interval_duration - self.rows - 1) * interval_duration
+        parameters: Dict[str, Union[int, str]] = {
             'pair': self.pair,
             'interval': interval_int,
+            'since': since,
         }
-        if self.since:
-            parameters['since'] = self.since
 
-        kraken = krakenex.API()
-        response = kraken.query_public('OHLC', data = parameters)
-        data = response.get('result', {}).get(self.pair, {})
-        self.data = pd.DataFrame(data, columns=self.columns)
+        response = requests.get(endpoint, params=parameters)
+        response.raise_for_status()
+
+        data_array = response.json().get('result', {}).get(self.pair.upper(), {})
+        data = pd.DataFrame(data_array, columns=KrakenOHLCV.columns)
+        data = data.head(self.rows).astype(KrakenOHLCV.dtypes)
+
+        self.__validate_data(data, response)
+
+        self.data = data
         return self.data
 
 
     def write(self, filepath: str):
-        self.data.to_csv(filepath)
+        self.data.to_csv(filepath, index=False)
 
```

Comment:
I realized that we should be using `index=False` for our write methods when writing unit tests. I'll add them to our other classes later.

Reply:
I can take care of that today.